### PR TITLE
Undefined Method for Collection Proxy

### LINKED ISSUE #33: Undefined Method for <ActiveRecord::Associations::CollectionProxy []>
I am seeing this issue on Rails `6.0.2.1`, Administrate `0.13.0` and the current `master` branch of NestedHasMany.

My configuration is very simple:

```
class LabDashboard < Administrate::BaseDashboard
  ATTRIBUTE_TYPES = {
    payor_lab_settings: Field::NestedHasMany.with_options(skip: :lab)
  }

  ...

  FORM_ATTRIBUTES = [
    :payor_lab_settings
  ].freeze
end

class PayorLabSettingDashboard < Administrate::BaseDashboard
  ATTRIBUTE_TYPES = {
    payor: Field::BelongsToSearch.with_options(class_name: 'Payor'),
  }.freeze
  
  ...

  FORM_ATTRIBUTES = [
    :payor
  ].freeze
end
```

Here is the relevant section of the stack trace:

```
PayorLabSetting::ActiveRecord_Associations_CollectionProxy#method_missing
activerecord (6.0.2.1) lib/active_record/relation/delegation.rb, line 108
Administrate::Page::Form#get_attribute_value
administrate (0.13.0) lib/administrate/page/base.rb, line 35
Administrate::Page::Form#attribute_field
administrate (0.13.0) lib/administrate/page/base.rb, line 29
block in Administrate::Page::Form#attributes
administrate (0.13.0) lib/administrate/page/form.rb, line 15
Administrate::Page::Form#attributes
administrate (0.13.0) lib/administrate/page/form.rb, line 14
Administrate::Field::NestedHasMany#nested_fields
() bundle/ruby/2.5.0/bundler/gems/administrate-field-nested_has_many-0145a35e4497/lib/administrate/field/nested_has_many.rb, line 23
```

The exact same code with `Field::HasMany` works correctly. All dashboards are working individually.

### PR DESCRIPTION
**Pull request by wkirby**
This fixes #33 by modifying the method of instantiating forms and setting data for the form builder. This also adds an explicit set of test for Administrate 0.13.0, which appears to have a number of breaking changes.

### CODE REVIEW

**Comment by efatsi** [comment]
Running into issues with this gem and administrate 0.13. Looks like this PR fixes the issues I'm seeing, would be great to get merged in!

**Comment by karensg** [comment]
Yes, please!

**Comment by LimeBlast** [comment]
I've gone ahead and referenced @wkirby's pull request in my `Gemfile`

```
gem 'administrate-field-nested_has_many', github: 'apsislabs/administrate-field-nested_has_many', branch: '33-undefined-method-for-collection-proxy'
```

Hopefully this will be OK until this gets merged in.

File: spec/dummy/config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  root to: redirect("/admin/schools")
```
Comment (sedubois) [inline spec/dummy/config/routes.rb]: I guess this is not needed as https://github.com/thoughtbot/administrate/pull/1574/files got merged

Comment (nickcharlton) [reply spec/dummy/config/routes.rb]: Yeah, we can drop 0585c69 from this branch.

**Comment by nickcharlton** [comment]
@wkirby, do you think you'd be able to drop that commit (0585c69) so we could merge this?

**Comment by wkirby** [comment]
@nickcharlton I've reverted 0585c69

**Comment by nickcharlton** [comment]
Thanks!

**Pull request merged**